core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestSimplifyExpressions.java

```diff
@@ -54,56 +61,101 @@
     @Test
     public void testPushesDownNegations()
     {
-        assertSimplifies("NOT X", "NOT X");
-        assertSimplifies("NOT NOT X", "X");
-        assertSimplifies("NOT NOT NOT X", "NOT X");
-        assertSimplifies("NOT NOT NOT X", "NOT X");
-
-        assertSimplifies("NOT (X > Y)", "X <= Y");
-        assertSimplifies("NOT (X > (NOT NOT Y))", "X <= Y");
-        assertSimplifies("X > (NOT NOT Y)", "X > Y");
-        assertSimplifies("NOT (X AND Y AND (NOT (Z OR V)))", "(NOT X) OR (NOT Y) OR (Z OR V)");
-        assertSimplifies("NOT (X OR Y OR (NOT (Z OR V)))", "(NOT X) AND (NOT Y) AND (Z OR V)");
-        assertSimplifies("NOT (X OR Y OR (Z OR V))", "(NOT X) AND (NOT Y) AND ((NOT Z) AND (NOT V))");
-
-        assertSimplifies("NOT (X IS DISTINCT FROM Y)", "NOT (X IS DISTINCT FROM Y)");
+        assertSimplifies("NOT X", "NOT X", ImmutableMap.of("X", BOOLEAN));
```

Comment:
Nice! Please move the change to existing tests to a commit before the main commit. It's a pre-requisite, but it's not directly related to the main change in this PR.